content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,444 @@
+
+If you are not seeing `datadog.apm.appsec_host` or `datadog.trace_agent.appsec.request`, check the [in-app instructions][3] to confirm that all steps for the initial setup are complete.
+
+## Application Security rate limits
+
+Application Security traces are rate-limited to 100 traces per second. Traces sent after the limit are not reported. Contact [Datadog support][1] if you need to change the limit.
+
+## Application Security events are missing in the dashboard
+
+There are a series of steps that must run successfully for Application Security events to appear in the Application Security dashboard. It is important to check each step when investigating this issue. Additional troubleshooting steps for specific languages are in the language tab at the end.
+
+### 1. Send a test attack to your application
+
+Simulate an attack to test your Application Security setup using the following command:
+
+```
+curl -A Arachni/v YOUR_APP_URL
+```
+
+View the resulting Application Security event in the [Application Security dashboard][4]. 
+
+It is possible that the test attack doesn't reach your application because of a firewall or another type of proxy server. It’s known that the following services block the Arachni test:
+
+- AWS API Gateway: Removes the user-agent and malicious parameters, even when its built-in WAF is deactivated
+- Extensible Service Proxy (ESP) by Google 
+
+To ensure that the Arachni test isn’t blocked:
+
+- Run the test from the machine hosting the service.
+- Verify that curl returns 200.
+- Verify that there is a trace with the attribute `@appsec.event:true`.
+- If [debug logs][5] are available, look for the following logs to see if the test is being blocked:
+
+  ```
+  DAS-0011-00: AppSec In-App WAF returned: <full_iaw_output> Took <exec_duration> ms
+
+  DDAS-0012-01: Detecting an attack from rule <rule_id>: <attack_info>
+
+  DDAS-0012-02: Blocked attack from <full_iaw_output>
+  ```
+
+### 2. Check if required tracer integrations are deactivated
+
+Application Security relies on certain tracer integrations. If they are deactivated, Application Security won't work. To see if there are deactivated integrations, look for `disabled_integrations` in your [startup logs][2].
+
+The required integrations vary by language.
+
+{{< programming-lang-wrapper langs="java,.NET,go,ruby,PHP,NodeJS" >}}
+{{< programming-lang lang="java" >}}
+
+For Java, the required integrations are the following:
+
+- servlet
+- servlet-2
+- servlet-3
+- servlet-request-body
+- grizzly
+- grizzly-char-body
+- grizzly-byte-body
+- grizzly-filterchain
+- spring-web
+- spring-webflux
+
+{{< /programming-lang >}}
+{{< programming-lang lang=".NET" >}}
+
+For ..NET, the required integrations are the following:
+
+- ASP.NET
+
+**Note:** If ASP.NET Core is disabled, Application Security should still work with this framework.
+
+{{< /programming-lang >}}
+{{< programming-lang lang="PHP" >}}
+
+**[NEED INFO]**
```

Comment:
Need PHP required integration info here.

Reply:
PHP AppSec doesn't require any integrations to be enabled, just the tracer itself.